src/main/java/org/folio/rest/impl/BatchVoucherExportsHelper.java

```diff
@@ -0,0 +1,113 @@
+package org.folio.rest.impl;
+
+import static org.folio.invoices.utils.HelperUtils.getEndpointWithQuery;
+import static org.folio.invoices.utils.HelperUtils.handleDeleteRequest;
+import static org.folio.invoices.utils.HelperUtils.handleGetRequest;
+import static org.folio.invoices.utils.HelperUtils.handlePutRequest;
+import static org.folio.invoices.utils.ResourcePathResolver.BATCH_VOUCHER_EXPORTS_STORAGE;
+import static org.folio.invoices.utils.ResourcePathResolver.resourceByIdPath;
+import static org.folio.invoices.utils.ResourcePathResolver.resourcesPath;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.rest.jaxrs.model.BatchVoucherExport;
+import org.folio.rest.jaxrs.model.BatchVoucherExportCollection;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
+
+import io.vertx.core.Context;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+
+public class BatchVoucherExportsHelper extends AbstractHelper {
+  private static final String GET_BATCH_VOUCHER_EXPORTS_BY_QUERY = resourcesPath(BATCH_VOUCHER_EXPORTS_STORAGE) + SEARCH_PARAMS;
+ 
+  public BatchVoucherExportsHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
+    super(okapiHeaders, ctx, lang);
+  }
+
+  /**
+   * Gets list of batch voucher exports
+   *
+   * @param limit  Limit the number of elements returned in the response
+   * @param offset Skip over a number of elements by specifying an offset value for the query
+   * @param query  A query expressed as a CQL string using valid searchable fields
+   * @return completable future with {@link BatchVoucherExportCollection} on success or an exception if processing fails
+   */
+  public CompletableFuture<BatchVoucherExportCollection> getBatchVoucherExports(int limit, int offset, String query) {
+    CompletableFuture<BatchVoucherExportCollection> future = new VertxCompletableFuture<>(ctx);
+    try {
+      String queryParam = getEndpointWithQuery(query, logger);
+      String endpoint = String.format(GET_BATCH_VOUCHER_EXPORTS_BY_QUERY, limit, offset, queryParam, lang);
+      handleGetRequest(endpoint, httpClient, ctx, okapiHeaders, logger).thenAccept(jsonVouchers -> {
+        logger.info("Successfully retrieved batch voucher exports: " + jsonVouchers.encodePrettily());
+        future.complete(jsonVouchers.mapTo(BatchVoucherExportCollection.class));
```

Comment:
This mapping might be blocking operation, i'd recommend using VertxCompletableFuture.supplyBlockingAsync or thenAccept(Apply)Async for mapping collections

Reply:
Done